src/launcher/features/initialisation/initialiseLauncherState.ts

```diff
@@ -109,37 +157,35 @@ const checkForLauncherUpdateAtStartup =
         }
     };
 
-const initialiseLauncherStateStage1 = (): AppThunk => async dispatch => {
-    dispatch(checkTelemetrySetting());
+const INTERRUPT_INITIALISATION = Symbol('interrupt initialisation');
 
-    await dispatch(loadSources());
-    await dispatch(loadApps());
-
-    dispatch(loadTokenInformation());
-};
-
-export const initialiseLauncherStateStage2 = (): AppThunk => dispatch => {
-    dispatch(downloadLatestAppInfoAtStartup());
-    dispatch(checkForLauncherUpdateAtStartup());
-
-    sendEnvInfo();
-};
+const initialisationActions = [
+    checkTelemetrySetting,
+    loadSources,
+    loadApps,
+    loadTokenInformation,
+    checkForDeprecatedSources,
+    checkForMissingTokenAndMigratedSources,
+    downloadLatestAppInfoAtStartup,
+    checkForLauncherUpdateAtStartup,
+    sendEnvInfo,
+];
 
-const checkForDeprecatedSources = (): AppThunk => (dispatch, getState) => {
-    const deprecatedSources = getSources(getState()).filter(isDeprecatedSource);
+const runRemainingInitialisationActionsSequentially =
+    (): AppThunk => async dispatch => {
+        while (initialisationActions.length > 0) {
+            const action = initialisationActions.shift() as () => AppThunk<
+                undefined | typeof INTERRUPT_INITIALISATION
+            >;
 
-    if (
-        deprecatedSources.length === 0 ||
-        getDoNotRemindDeprecatedSources(getState())
-    ) {
-        dispatch(initialiseLauncherStateStage2());
-    } else {
-        dispatch(showDeprecatedSources(deprecatedSources));
-    }
-};
+            const result = await dispatch(action()); // eslint-disable-line no-await-in-loop -- Must be awaited because some actions are asynchronous
 
-export default (): AppThunk => async dispatch => {
-    await dispatch(initialiseLauncherStateStage1());
+            if (result === INTERRUPT_INITIALISATION) {
+                return;
+            }
+        }
+    };
```

Comment:
Even though I am not a fan of this kind of function list, but it's a really clean solution. I like it a lot!